### DISCUSSION TITLE
VSC workspace recommended extensions

### DISCUSSION THREAD

**Discussion by talmurshidi** [HackYourFutureBelgium/template-markdown#4] [Ideas]
## Motivation
Tools consistency are very important in any development team to have a smooth work. HYF uses VSC as main tool for web development.
VSC has many extensions to be installed before start developing a project. In some cases the learners don't have the save extensions installed in VSC.

## Solution
VSC offers [workspace-recommended-extensions](https://code.visualstudio.com/docs/editor/extension-marketplace#_workspace-recommended-extensions). Creating a file `extensions.json` inside `.vscode` folder.
An example extensions.json could be:
```
{
  "recommendations": ["dbaeumer.vscode-eslint", "esbenp.prettier-vscode"]
}
```

## Recommended Extensions

1. [Stylelint](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint)
2. [HTML End Tag Labels](https://marketplace.visualstudio.com/items?itemName=anteprimorac.html-end-tag-labels)
3. [Bookmarks](https://marketplace.visualstudio.com/items?itemName=alefragnani.Bookmarks)
4. <del>[Auto Rename Tag](https://marketplace.visualstudio.com/items?itemName=formulahendry.auto-rename-tag)</del> (integrated in VSC)
5. [Project Manager](https://marketplace.visualstudio.com/items?itemName=alefragnani.project-manager)
6. [GitHub Pull Requests and Issues](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github)
7. [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer)
8. [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
9. [GitLens — Git supercharged](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens)
10. [Git History](https://marketplace.visualstudio.com/items?itemName=donjayamanne.githistory)
11. [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)
12. [vscode-icons](https://marketplace.visualstudio.com/items?itemName=vscode-icons-team.vscode-icons)
13. [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)
14. [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)
15. [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
16. [Git Graph](https://marketplace.visualstudio.com/items?itemName=mhutchie.git-graph)

### Update 29/04/2023
17. [Markdown Preview Github Styling](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-preview-github-styles)
18. [HTML CSS Support](https://marketplace.visualstudio.com/items?itemName=ecmel.vscode-html-css)
19. [HTMLHint](https://marketplace.visualstudio.com/items?itemName=HTMLHint.vscode-htmlhint)
20. [CSS Peek](https://marketplace.visualstudio.com/items?itemName=pranaygp.vscode-css-peek)
21. [IntelliSense for CSS class names in HTML](https://marketplace.visualstudio.com/items?itemName=Zignd.html-css-class-completion)
22. <del>[indent-rainbow](https://marketplace.visualstudio.com/items?itemName=oderwat.indent-rainbow)</del> (integrated in VSC)
23. [Dependency Cruiser Extension](https://marketplace.visualstudio.com/items?itemName=juanallo.vscode-dependency-cruiser)
24. <del>[Blockman - Highlight Nested Code Blocks](https://marketplace.visualstudio.com/items?itemName=leodevbro.blockman)</del>
25. <del>[Bracket Pair Colorizer](https://marketplace.visualstudio.com/items?itemName=CoenraadS.bracket-pair-colorizer-2)</del> (integrated in VSC)
26. [IntelliCode](https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.vscodeintellicode)
27. [Better Comments](https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments)
28. [Path Intellisense](https://marketplace.visualstudio.com/items?itemName=christian-kohler.path-intellisense)
29. [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare)

## No Need Extensions (to be checked)
- [VS Code - You don't need that extension](https://www.roboleary.net/vscode/2020/08/05/dont-need-extensions.html)
- [VS Code: You don't need that extension part 2](https://www.roboleary.net/2021/11/06/vscode-you-dont-need-that-extension2.html)
- [VS Code - You don't need a formatting extension (Prettier and friends)](https://www.roboleary.net/tools/2022/05/18/vscode-you-dont-need-a-formatting-extension-prettier-and-friends.html)

**Comment by talmurshidi** [comment]
```
{
	// See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
	// Extension identifier format: ${publisher}.${name}. Example: vscode.csharp

	// List of extensions which should be recommended for users of this workspace.
	"recommendations": [
		"streetsidesoftware.code-spell-checker",
		"dbaeumer.vscode-eslint",
		"mhutchie.git-graph",
		"donjayamanne.githistory",
		"github.vscode-pull-request-github",
		"eamodio.gitlens",
		"anteprimorac.html-end-tag-labels",
		"ritwickdey.liveserver",
		"yzhang.markdown-all-in-one",
		"davidanson.vscode-markdownlint",
		"esbenp.prettier-vscode",
		"alefragnani.project-manager",
		"stylelint.vscode-stylelint",
		"vscode-icons-team.vscode-icons",
		"formulahendry.auto-rename-tag",
		"alefragnani.bookmarks"
	],
	// List of extensions recommended by VS Code that should not be recommended for users of this workspace.
	"unwantedRecommendations": [
		
	]
}
```

**Comment by colevandersWands** [comment]
There's also the [list of plugins from the precourse](https://github.com/DeNepo/precourse/blob/master/deliverables.md#3-install-vscode-extensions).  Let's also update that list to include everything here?

Then the VSCode config can be a backup for HYF students, and a full check for anyone else using it.

**Reply by talmurshidi** [reply]
Thank you for the list of plugins, I have forgot it :)
Sure, I will update the list and keep it consistency.
The idea of `extensions.json` file, it shows a prompt to the user to install the extension in most cases.
